includes/templates/template_default/jscript/zen_addr_pulldowns.php

```diff
@@ -11,47 +11,46 @@
  * @version $Id: Zen4All 2019 Jun 03 Modified in v1.5.7 $
  */
 // -----
-// If the site does NOT require a 'State' entry in an address or has NOT enabled states to
-// be displayed as dropdowns, nothing to be done here!
+// If the site does NOT require a 'State' entry in an address, nothing to be done here!
 //
-if (ACCOUNT_STATE !== 'true' || ACCOUNT_STATE_DRAW_INITIAL_DROPDOWN !== 'true') {
+if (ACCOUNT_STATE !== 'true') {
     return;
 }
 
 // -----
-// If the current site has at least one country enabled that uses zones, a JSON-encoded array of
-// countries-to-zones will be created for use by the jQuery section.
+// If the current site is using the state dropdowns, create the array that identifies the various
+// zones for the currently-active countries.  When state dropdowns aren't being used, the
+// empty $c2z array will 'instruct' the jQuery section to presume that no countries have
+// associated zones.
 //
-$countries = $db->Execute(
-    "SELECT DISTINCT zone_country_id
-       FROM " . TABLE_ZONES . "
-            INNER JOIN " . TABLE_COUNTRIES . "
-                ON countries_id = zone_country_id
-               AND status = 1
-   ORDER BY zone_country_id"
-);
-
 $c2z = [];
-foreach ($countries as $next_country) {
-    $current_country_id = $next_country['zone_country_id'];
-    $c2z[$current_country_id] = [];
-
-    $states = $db->Execute(
-        "SELECT zone_name, zone_id
+if (ACCOUNT_STATE_DRAW_INITIAL_DROPDOWN === 'true') {
+    // -----
+    // If the current site has at least one country enabled that uses zones, a JSON-encoded array of
+    // countries-to-zones will be created for use by the jQuery section.
+    //
+    $countries = $db->Execute(
+        "SELECT DISTINCT zone_country_id
            FROM " . TABLE_ZONES . "
-          WHERE zone_country_id = $current_country_id
-       ORDER BY zone_name"
+                INNER JOIN " . TABLE_COUNTRIES . "
+                    ON countries_id = zone_country_id
+                   AND status = 1
+       ORDER BY zone_country_id"
     );
-    foreach ($states as $next_state) {
-        $c2z[$current_country_id][$next_state['zone_id']] = $next_state['zone_name'];
-    }
-}
+    foreach ($countries as $next_country) {
+        $current_country_id = $next_country['zone_country_id'];
+        $c2z[$current_country_id] = [];
 
-// -----
-// If none of the currently-enabled countries use zones, nothing to do here.
-//
-if (count($c2z) === 0) {
-    return;
+        $states = $db->Execute(
```

Comment:
this section of code looks very much like the function, `zen_get_country_zones`.  could we not use that function here as opposed to replicating code?  in addition, i am of the belief that we should always include the `zone_code` whenever we are selecting the `zone_id` from this table.  if for no other reason than to appease developers like myself that want to use said field in future development and do not have to jump through hoops to get it.

Reply:
Even though it goes against my "performance-grain", I'm amenable to making the change to use `zen_get_country_zones`.

I'll grant that using the `zone_code` would be preferable, but that wouldn't be downward-compatible with the base Zen Cart templates or any add-on templates in existence.

Reply:
i'm not sure i understand your "performance-grain" comment.

in addition, i think i was perhaps mis-understood.  i understand your concerns about use of `zone-code`.  my comment was that the function, `zen_get_country_zones` should also the return the `zone_code` in the return array; despite it not being used here.  it would not affect anything done here.  i can do a PR for said change, if you are amenable.

Reply:
I was just referring to the double iteration of the zones' information.  Having `zen_get_country_zones` return the `zone_code` is a totally different 'animal' and should be addressed in a separate issue.